packages/editor/src/components/post-actions/actions.js

```diff
@@ -307,34 +307,29 @@ const trashPostAction = {
 	},
 };
 
-function useTrashPostAction( postType ) {
+function useCanUserEligibilityCheckPostType( capability, resource, action ) {
 	const registry = useRegistry();
-	const { resource, cachedCanUserResolvers } = useSelect(
-		( select ) => {
-			const { getPostType, getCachedResolvers } = select( coreStore );
-			return {
-				resource: getPostType( postType )?.rest_base || '',
-				cachedCanUserResolvers: getCachedResolvers().canUser,
-			};
-		},
-		[ postType ]
-	);
 	return useMemo(
 		() => ( {
-			...trashPostAction,
+			...action,
 			isEligible( item ) {
 				return (
-					trashPostAction.isEligible( item ) &&
+					action.isEligible( item ) &&
 					registry
 						.select( coreStore )
-						.canUser( 'delete', resource, item.id )
+						.canUser( capability, resource, item.id )
```

Comment:
I guess it doesn't really happen in the case, but the problem here is that `isEligible` will not be called if the user capability changes, right? So maybe all of these need to be precalculated/subscribed to at some point so the buttons re-render.